src/makeInfinityAdder.js

```diff
@@ -6,7 +6,25 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const adds = (num) => {
+    if (num !== undefined) {
+      sum += num;
+    }
+
+    if (num === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
```

Comment:
```suggestion
    if (num === undefined) {
      const result = sum;

      sum = 0;

      return result;
    }
    
    sum += num;
```
Let's swap these statements by places and get rid of redundant `if`